### PR TITLE
feat(Run on complete solution): Run stryker on all projects in a solution

### LINKED ISSUE #1379: Solution runs are inconsistant
**Describe the bug**
I have seen a project getting different stryker results every time I ran it. It happens when running the full solution.

Buildalyzer seemed to find less test projects than expected on some runs. 

**Expected behavior**
I expect a consistant result every run.

**Desktop (please complete the following information):**
 - OS: Windows
 - Type of project core 3.1
 - Framework Version NET 5.0
 - Stryker Version 0.21.0

**Additional context**
The solution run feature is still experimental.

### PR DESCRIPTION
**Pull request by richardwerkman**
To do:
- [x] Add integration test
- [x] Test on some large projects
- [x] Test on project with diferent types of projects (framework/core)
- [ ] Add unit tests

closes #1379 and #1323

### CODE REVIEW

File: integrationtest/TargetProjects/ConsoleApp1/ConsoleApp1.csproj [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (rouke-broersma) [inline integrationtest/TargetProjects/ConsoleApp1/ConsoleApp1.csproj:1]: Should probably give this a better name

File: integrationtest/TargetProjects/IntegrationTestApp.sln [hunk 2]
```diff
@@ -0,0 +1,64 @@
+﻿
```
Comment (rouke-broersma) [inline integrationtest/TargetProjects/IntegrationTestApp.sln:1]: Don't we already have a solution file the the integration test project?

File: integrationtest/TargetProjects/Library/Library.csproj [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net6</TargetFramework>
```
Comment (rouke-broersma) [inline integrationtest/TargetProjects/Library/Library.csproj:4]: How do we guarantee backwards compatibility if we no longer test dotnet core

File: src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs [hunk 4]
```diff
@@ -26,6 +26,12 @@ public class StrykerOptions
         /// </summary>
         public string ProjectPath { get; init; }
 
+        /// <summary>
+        /// When true, stryker is mutating all projects in a solution
+        /// </summary>
+        /// <returns></returns>
+        public bool IsSolutionContext => SolutionPath != null && FilePathUtils.NormalizePathSeparators(ProjectPath) == FilePathUtils.NormalizePathSeparators(Path.GetDirectoryName(SolutionPath));
```
Comment (rouke-broersma) [inline src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs:33]: This is not an option, it does not belong here

File: src/Stryker.Core/Stryker.Core/Initialisation/ProjectInfo.cs [hunk 5]
```diff
@@ -30,11 +30,11 @@ public class ProjectInfo
...
-            foreach (var testProject in TestProjectAnalyzerResults)
-            {
-                var injectionPath = GetInjectionFilePath(testProject);
-                _fileSystem.File.Copy(GetBackupName(injectionPath), injectionPath, true);
-            }
+            //foreach (var testProject in TestProjectAnalyzerResults)
+            //{
+            //    var injectionPath = GetInjectionFilePath(testProject);
+            //    _fileSystem.File.Copy(GetBackupName(injectionPath), injectionPath, true);
+            //}
```
Comment (rouke-broersma) [inline src/Stryker.Core/Stryker.Core/Initialisation/ProjectInfo.cs:37]: ?

File: src/Stryker.Core/Stryker.Core/Initialisation/ProjectOrchestrator.cs [hunk 6]
```diff
@@ -25,20 +26,23 @@ public class ProjectOrchestrator : IProjectOrchestrator
...
         }
 
         public IEnumerable<IMutationTestProcess> MutateProjects(StrykerOptions options, IReporter reporters)
         {
-            if (IsSolutionContext(options))
+            if (options.IsSolutionContext)
             {
-                // Build and analyze all projects in the solution with buildalyzer
+                // Analyze all projects in the solution with buildalyzer
                 var solutionProjects = AnalyzeSolution(options);
```
Comment (rouke-broersma) [inline src/Stryker.Core/Stryker.Core/Initialisation/ProjectOrchestrator.cs:46]:
```suggestion
                var solutionAnalyzerResults = AnalyzeSolution(options);
```

We need a better name for this, all throughout. solutionProjects is confusing, are they ProjectComponents? ProjectInfo?

File: integrationtest/TargetProjects/IntegrationTestApp.sln (hunk 2 above)
Comment (richardwerkman) [reply integrationtest/TargetProjects/IntegrationTestApp.sln:1]: We have one including the validation tests. But now I wanted one with just the target project to run solution runs against.

File: integrationtest/TargetProjects/Library/Library.csproj (hunk 3 above)
Comment (richardwerkman) [reply integrationtest/TargetProjects/Library/Library.csproj:4]: Good point. I'll see if we can keep some on 3.1

File: src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs (hunk 4 above)
Comment (richardwerkman) [reply src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs:33]: It kinda is an option. The user decides if the run is a solution run or not. They don't specify it by passing a value but by the working directory they run from. I don't know a better place to put this

File: src/Stryker.Core/Stryker.Core/Initialisation/ProjectInfo.cs (hunk 5 above)
Comment (richardwerkman) [reply src/Stryker.Core/Stryker.Core/Initialisation/ProjectInfo.cs:37]: Oops, was for testing perposes

File: integrationtest/TargetProjects/ConsoleApp1/ConsoleApp1.csproj (hunk 1 above)
Comment (richardwerkman) [reply integrationtest/TargetProjects/ConsoleApp1/ConsoleApp1.csproj]: Removed whole project as it was unused

File: src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs (hunk 4 above)
Comment (richardwerkman) [reply src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs]: I know you don't agree with this solution. But can we create a new issue to fix the size of the options class? I feel this PR is big enough as it is

File: integrationtest/TargetProjects/IntegrationTestApp.sln (hunk 2 above)
Comment (richardwerkman) [reply integrationtest/TargetProjects/IntegrationTestApp.sln:1]: Do you agree with this? Or else we can add a new issue to clean up the integration test project as it is a bit of a mess right now :(

Comment (rouke-broersma) [reply integrationtest/TargetProjects/IntegrationTestApp.sln:1]: I think that's good in any case

**Pull request merged**